### PR TITLE
test: 100% coverage for `timeAgo`

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Achieve 100% coverage in `timeAgo`.

- remove singular verbose “second ago”
- remove singular “day ago”
- add test for future dates

## Why?

[Greener is better!](https://coveralls.io/builds/40626836/source?filename=src%2FtimeAgo.ts)

### CODE REVIEW

File: src/timeAgo.test.ts
Comment (oliverlloyd) [inline src/timeAgo.test.ts:227]: 👍 nice

Comment (mxdvl) [reply src/timeAgo.test.ts:227]: `false` 😆

File: src/timeAgo.ts
Comment (oliverlloyd) [inline src/timeAgo.ts:67]: Would this not allow the result "Last updated 1 days ago"?

Comment (mxdvl) [reply src/timeAgo.ts:67]: It would return “Last updated yesterday 10:00”. We get hours up to 1 day ago, then “Yesterday HH:MM” up to “two days ago”.

Comment (oliverlloyd) [reply src/timeAgo.ts:67]: Ah ha! Great spot

**Review by oliverlloyd — APPROVED** [review]

**Comment by mxdvl** [comment]
Ta, I need a +1 from a `CODEOWNER`, too.

Also: I was investigating this a little more, and wondered what happens during BST…

File: src/timeAgo.ts
Comment (sndrs) [reply src/timeAgo.ts:67]: do you fancy adding a comment explaining this? and the same with seconds above (<15 = 'now')?

File: src/timeAgo.test.ts
Comment (sndrs) [reply src/timeAgo.test.ts:227]: should it return `false` or should it error?

**Review by sndrs — CHANGES_REQUESTED** [review]

**Comment by oliverlloyd** [comment]
> should it return false or should it error?

Or perhaps it could return 'now' for future dates? I think a date in the future is at least theoretically possible, it's hard to keep server time in sync, I've definitely seen thing like "Last updated in 2 minutes" on other platforms. So maybe we want to return the most useful response for the most likely scenario, where the date is a few seconds in the future?

**Comment by mxdvl** [comment]
Or return “recently” and warn inside the console? It’s a soft fallback that isn’t wrong for the user.

**Comment by oliverlloyd** [comment]
> Or return “recently” and warn inside the console? It’s a soft fallback that isn’t wrong for the user, while remaining distinctly different to other values.

I like how this provides a graceful failover in all cases 👍 . I guess the question is, do we want to trigger errors on future dates so that we discover them or handle them silently?

File: src/timeAgo.test.ts
Comment (mxdvl) [reply src/timeAgo.test.ts:227]: > Or perhaps it could return 'now' for future dates? I think a date in the future is at least theoretically possible, it's hard to keep server time in sync, I've definitely seen thing like "Last updated in 2 minutes" on other platforms. So maybe we want to return the most useful response for the most likely scenario, where the date is a few seconds in the future? -@oliverlloyd 
>
> Or return “recently” and warn inside the console? It’s a soft fallback that isn’t wrong for the user, while remaining distinctly different to other values. -@mxdvl
>
> I like how this provides a graceful failover in all cases 👍 . I guess the question is, do we want to trigger errors on future dates so that we discover them or handle them silently?- @oliverlloyd 

I’m thinking we should handle silently for now? I don’t think an error in a timezones should stop JS from executing.

Comment (sndrs) [reply src/timeAgo.test.ts:227]: it may be beyond the scope of this PR, but perhaps this module should be called `timeFromNow` or `toRelativeTime` and handle past and future dates?

**Review by sndrs — APPROVED** [review]
great!

File: src/timeAgo.test.ts
Comment (mxdvl) [reply src/timeAgo.test.ts:227]: I think the final solution to this can be addressed in a subsequent PR :D

**Pull request merged**
